src/t_stream.c

```diff
@@ -2031,21 +2031,36 @@ void xclaimCommand(client *c) {
             force = 1;
         } else if (!strcasecmp(opt,"JUSTID")) {
             justid = 1;
-        } else if (!strcasecmp(opt,"IDLE") && moreargs) {
+        } else if (!strcasecmp(opt,"IDLE")) {
+            if (!moreargs) {
+                addReplyErrorFormat(c,"Lack of IDLE option argument for XCLAIM");
+                return;
+            }
+
             j++;
             if (getLongLongFromObjectOrReply(c,c->argv[j],&deliverytime,
                 "Invalid IDLE option argument for XCLAIM")
                 != C_OK) return;
             deliverytime = now - deliverytime;
-        } else if (!strcasecmp(opt,"TIME") && moreargs) {
+        } else if (!strcasecmp(opt,"TIME")) {
+            if (!moreargs) {
```

Comment:
I understand what's the goal here, but I think we should follow the other Redis commands and just provide a syntax error here, without making the code more complex just to specify where the error is. It's trivial for the user to spot the error anyway in case of misuse of the option.

Reply:
@antirez OK, i have reset this and fix a minor error log.